lib/validators/package-validator.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Validates if the component set package size
+ */
+
+import { ComponentSet, GetFileContentType } from '../models';
+import { Validator } from './validator';
+
+const MAX_COMPONENT_SET_FILE_COUNT = 5000;
+const MAX_COMPONENT_SET_SIZE_MB = 100;
+const MAX_CUSTOM_DATA_FILE_COUNT = 1000;
+const MAX_CUSTOM_DATA_SIZE_MB = 20;
+const MB_IN_BYTES = 1 * 1000 * 1000;
+
+interface FileInfo {
+    path: string;
+    size: number;
+}
+
+export class PackageValidator extends Validator {
+    constructor(
+        error: (errorMessage: string) => false,
+        definition: ComponentSet,
+        private filePaths: Set<string>,
+        private getFileContent: GetFileContentType,
+    ) {
+        super(error, definition);
+    }
+
+    async validate(): Promise<void> {
+        const files: FileInfo[] = [];
+        for (let filePath of Array.from(this.filePaths)) {
+            const file = (await this.getFileContent(filePath)) as Buffer | string;
```

Comment:
Should we implement a new function next to getFileContent to get the file size? Loading the content into memory is not the way to go I think. `fs` likely has a functions to get the file size statistics (`stats`) and in the backend we might be able to use the information from the zip file.

Reply:
Done